### PR TITLE
Enhanced RDoc for strftime

### PR DESCRIPTION
**Pull request by BurdetteLamar**
Creates a single discussion for strftime formats that three methods can link to.

Treats:
- Time#strftime.
To be linked later:
- DateTime#strftime.
- Date#strftime.

I could not find a Rakefile where I can add the new file to an RDoc list.

### CODE REVIEW

File: doc/time/strftime.rdoc [hunk 1]
```diff
@@ -0,0 +1,353 @@
...
+<b>Minumum Field Width</b>
+
+The default width for a field is the width of the data itself.
+The integer optional minimum field width specifies the minimum width;
+it is ignored if smaller than the default width:
+
+  t = Time.now
+  t.strftime('%Y')   # => "2021"
+  t.strftime('%8Y')  # => "00002021"
+  t.strftime('%20Y') # => "2021"
```
Comment (mame) [inline doc/time/strftime.rdoc:37]: Do you mean `%0Y`?

File: doc/time/strftime.rdoc [hunk 2]
```diff
@@ -0,0 +1,353 @@
...
+  t.strftime('%08Y')    # => "00002021"
+
+  t.strftime('%-6b')    # => "Oct"
+  t.strftime('%_6b')    # => "   Oct"
+  t.strftime('%06b')    # => "000Oct"
+
+Flags for casing:
+
+- <tt>^</tt>: Upcase.
+- <tt>%</tt>: Change case.
```
Comment (mame) [inline doc/time/strftime.rdoc:64]: `#`

File: doc/time/strftime.rdoc [hunk 3]
```diff
@@ -0,0 +1,353 @@
...
+
+<b>Conversion Specifiers</b>
+
+Note: Although technically a conversion specifier is a single character,
+we follow tradition here and list each with a prefixed percent character.
+
+<em>Year</em>
+
+- <tt>%Y</tt>: Year with century; may be negative; at least four digits.
+- <tt>%C</tt>: 2-digit century; <tt>year / 100</tt>, rounded down.
```
Comment (mame) [inline doc/time/strftime.rdoc:106]: Not necessarily 2-digit.

File: doc/time/strftime.rdoc [hunk 4]
```diff
@@ -0,0 +1,353 @@
...
+  t.strftime('%C')    # => "20"
+
+  t  = Time.now       # => 2021-11-01 09:21:41.0455507 -0500
+  t.strftime('%y')    # => "21"
+
+<em>Month</em>
+
+- <tt>%m</tt>: 2-digit month of the year, zero-padded; range (1..12).
+- <tt>%B</tt>: Full month name.
+- <tt>%b</tt>: 3-character abbreviated month name,
```
Comment (mame) [inline doc/time/strftime.rdoc:130]: Why did you remove some examples such as `%_m` and `%^B`?

I like the previous document that includes an example output in line, like `<tt>%B</tt>: Full month name. (``January'')`

File: doc/time/strftime.rdoc [hunk 5]
```diff
@@ -0,0 +1,353 @@
...
+The ISO year begins with the first Monday that is or precedes January 4.
+
+- <tt>%G</tt>: ISO 8601 year.
+- <tt>%g</tt>: 2-digit ISO 8601 year; range (0..99).
+
+Examples:
+
+  t.strftime('%G') # => "2021"
+  t.strftime('%g') # => "21"
+  t.strftime('%V') # => "43"
```
Comment (mame) [inline doc/time/strftime.rdoc:289]: This line should be removed

File: doc/time/strftime.rdoc [hunk 6]
```diff
@@ -0,0 +1,353 @@
...
+
+- <tt>%V</tt>: ISO week number; range (1..53).
```
Comment (mame) [inline doc/time/strftime.rdoc:291]: The following document is missing?

>  The week 1 of YYYY starts with a Monday and includes YYYY-01-04.
>  The days in the year before the first week are in the last week of
>  the previous year.

File: ext/date/date_core.c [hunk 7]
```diff
@@ -6790,178 +6790,11 @@ date_strftime_internal(int argc, VALUE *argv, VALUE self,
...
- * locale independent representations, %D and %T.
- *
- * Examples:
- *
- *    d = DateTime.new(2007,11,19,8,37,48,"-06:00")
- *				#=> #<DateTime: 2007-11-19T08:37:48-0600 ...>
- *    d.strftime("Printed on %m/%d/%Y")   #=> "Printed on 11/19/2007"
- *    d.strftime("at %I:%M%p")            #=> "at 08:37AM"
- *
- * Various ISO 8601 formats:
```
Comment (mame) [inline ext/date/date_core.c:6928]: This example table is very useful for me as a cheetsheet. It should be kept.

File: ext/date/date_core.c [hunk 8]
```diff
@@ -6790,178 +6790,11 @@ date_strftime_internal(int argc, VALUE *argv, VALUE self,
...
- *
- * This method is similar to the strftime() function defined in ISO C
- * and POSIX.
- * Several directives (%a, %A, %b, %B, %c, %p, %r, %x, %X, %E*, %O* and %Z)
- * are locale dependent in the function.
- * However, this method is locale independent.
- * So, the result may differ even if the same format string is used in other
- * systems such as C.
- * It is good practice to avoid %x and %X because there are corresponding
- * locale independent representations, %D and %T.
```
Comment (mame) [inline ext/date/date_core.c:6919]: This note about locale is missing?

**Comment by mame** [comment]
When you are going to rewrite the document in a major way, please be careful not to delete any description. If you really want to delete something intentionally, please create a separete PR and explain why.

File: doc/time/strftime.rdoc (hunk 3 above)
Comment (nobu) [reply doc/time/strftime.rdoc:106]: It is zero-filled 2-digits at least.

**Comment by nobu** [comment]
`ext/date` is a separate library, and may not be completely same as `Time`.

File: doc/time/strftime.rdoc (hunk 1 above)
Comment (BurdetteLamar) [reply doc/time/strftime.rdoc]: Yes, now corrected.  Also added a negative-integer example.

File: doc/time/strftime.rdoc (hunk 3 above)
Comment (BurdetteLamar) [reply doc/time/strftime.rdoc]: Corrected, and examples added.

File: doc/time/strftime.rdoc (hunk 4 above)
Comment (BurdetteLamar) [reply doc/time/strftime.rdoc:138]: _ and ^ apply to many specifiers, not just these few.  I did add a note in Flags about multiple flags.

'example output in line':  There were 10 of these in all.  Will anyone misunderstand 'Full month name'?

File: doc/time/strftime.rdoc (hunk 5 above)
Comment (BurdetteLamar) [reply doc/time/strftime.rdoc]: Removed.

File: doc/time/strftime.rdoc (hunk 6 above)
Comment (BurdetteLamar) [reply doc/time/strftime.rdoc:302]: Restored.

File: ext/date/date_core.c (hunk 7 above)
Comment (BurdetteLamar) [reply ext/date/date_core.c:6928]: Restored.

File: ext/date/date_core.c (hunk 8 above)
Comment (BurdetteLamar) [reply ext/date/date_core.c:6919]: Restored.

**Comment by BurdetteLamar** [comment]
> `ext/date` is a separate library, and may not be completely same as `Time`.

I've noted six differences among the specifiers.  For those I've given examples for all three classes, showing the different outputs.  Have I missed other differences?

**Comment by BurdetteLamar** [comment]
> `ext/date` is a separate library, and may not be completely same as `Time`.

Also means that any mods must be made over at ruby/date, not here in ruby/ruby?

**Comment by BurdetteLamar** [comment]
I have restored ext/date/date_core.c, which should not be modified here.

**Comment by BurdetteLamar** [comment]
@mame and @nobu, are there problems here?

File: ext/date/date_core.c (hunk 8 above)
Comment (BurdetteLamar) [reply ext/date/date_core.c]: @sven-frotscher, I guess I don't understand.   This PR is closed.

Comment (duerst) [reply ext/date/date_core.c]: > Since issues are disabled in this repo: Please, Ruby, respect my environment variables and localize the output of `strftime()`...

@sven-frotscher: If you have an issue, please report it at https://bugs.ruby-lang.org/projects/ruby-master.

Comment (sven-frotscher) [reply ext/date/date_core.c]: > > Since issues are disabled in this repo: Please, Ruby, respect my environment variables and localize the output of `strftime()`...
> 
> @sven-frotscher: If you have an issue, please report it at https://bugs.ruby-lang.org/projects/ruby-master.

I don't use Ruby nearly enough to justify creating an account in its issue tracker, sorry. Though if someone else has one and wants to report this issue, feel free to do so. I've worked around it by porting my project to another scripting language.
